redis/src/main/scala/zio/redis/TestExecutor.scala

```diff
@@ -79,21 +86,285 @@ private[redis] final class TestExecutor private (
+              .map(bool => RespValue.Integer(if (bool) 1 else 0))
+          } else
+            STM.succeedNow {
+              RespValue.Integer(0)
+            }
+        }
+
+      case api.Connection.ClientList =>
+        val inputList  = input.toList.map(_.asString)
+        val typeOption = if (inputList.contains("TYPE")) Some(inputList(1)) else None
+        val inputIds   = if (inputList.contains("ID")) inputList.tail else List[String]()
+
+        for {
+          flags <- clientTrackingInfo.get("flags")
+          id    <- clientInfo.get("id")
+          info <-
+            if (
+              (typeOption.isEmpty || flags.get.contains(typeOption.get.toLowerCase))
+              && (inputIds.isEmpty || inputIds.contains(id.get))
+            ) clientInfo.fold("")((z, a) => s"$z ${a._1}=${a._2}")
+            else STM.succeed("")
+        } yield RespValue.bulkString(info)
+
+      case api.Connection.ClientGetName =>
+        for {
+          name <- clientInfo.get("name")
+        } yield name.map(RespValue.bulkString).getOrElse(RespValue.NullBulkString)
+
+      case api.Connection.ClientGetRedir =>
+        clientTrackingInfo.get("redirect").map(redir => RespValue.Integer(redir.get.head.toLong))
+
+      case api.Connection.ClientUnpause =>
+        STM.succeedNow {
+          RespValue.SimpleString("OK")
+        }
+
+      case api.Connection.ClientPause =>
+        STM.succeedNow {
+          RespValue.SimpleString("OK")
+        }
+
+      case api.Connection.ClientSetName =>
+        val name = input.head.asString
+        for {
+          _ <- clientInfo.put("name", name)
+        } yield RespValue.SimpleString("OK")
+
+      case api.Connection.ClientTracking =>
+        val inputList = input.toList.map(_.asString)
+
+        if (inputList.head == "ON") {
+          val mode = inputList match {
+            case list if list.contains("OPTIN")  => Some("optin")
+            case list if list.contains("OPTOUT") => Some("optout")
+            case list if list.contains("BCAST")  => Some("bcast")
+            case _                               => None
+          }
+          val noLoop = inputList.contains("NOLOOP")
+          val filteredInput = inputList.tail
+            .filter(input => input != "OPTIN" && input != "OPTOUT" && input != "BCAST" && input != "NOLOOP")
+            .grouped(2)
+            .toList
+          val redirectId = filteredInput.filter(_.head == "REDIRECT").map(_(1)).headOption
+          val prefixes   = filteredInput.filter(_.head == "PREFIX").map(_(1)).toSet
+
+          if (prefixes.isEmpty || mode.nonEmpty && mode.get == "bcast") {
+            val addTracking = for {
+              flags   <- clientInfo.get("flags")
+              newFlags = if (!flags.get.contains('t')) flags.get + 't' else flags.get
+              _       <- clientInfo.put("flags", newFlags)
+              _       <- clientTrackingInfo.put("flags", Set("on"))
+            } yield STM.succeed(())
```

Comment:
```suggestion
            } yield STM.unit
```
Also, there's no need for nested effects here (i.e. `STM[STM[Unit]]`). It is useful in certain situations, but it can lead to "swallowed" operations unless it's properly flatten. In this case it's sufficient to return `()`. The same remark applies for all subsequent for comprehensions.